src/http2_connection.erl

```diff
@@ -1541,22 +1562,25 @@ good_req_headers(Headers) ->
     end.
 
 no_upper_names(Headers) ->
+    lager:error("Headers: ~p", [Headers]),
     lists:all(
       fun({Name,_}) ->
               NameStr = binary_to_list(Name),
               NameStr =:= string:to_lower(NameStr)
       end,
      Headers).
 
-validate_pseudos([{<<":path">>,_V}|Tail]) ->
-    validate_pseudos(Tail);
-validate_pseudos([{<<":method">>,_V}|Tail]) ->
-    validate_pseudos(Tail);
-validate_pseudos([{<<":scheme">>,_V}|Tail]) ->
-    validate_pseudos(Tail);
-validate_pseudos([{<<":authority">>,_V}|Tail]) ->
-    validate_pseudos(Tail);
-validate_pseudos(DoneWithPseudos) ->
+validate_pseudos(request, [{<<":path">>,_V}|Tail]) ->
```

Comment:
correct. This test, and #40 which this also fixes, are only about failing when incorrect pseudoheaders are encountered.

Actually, it doesn't seem like the `h2spec` tool tests for "missing required pseudoheader", so I'll make note to figure out what that means and include another test for it.